File: app/src/main/java/com/example/shopstock/ProductDB.java
package com.example.shopstock;

import androidx.annotation.NonNull;
import androidx.room.Database;
import androidx.room.RoomDatabase;
import androidx.room.migration.Migration;
import androidx.sqlite.db.SupportSQLiteDatabase;

@Database(entities = {Product.class}, version = 3) // Update version number
public abstract class ProductDB extends RoomDatabase {
    public abstract ProductDao productDao();

    public static final Migration MIGRATION_1_2 = new Migration(1, 2) {
        @Override
        public void migrate(@NonNull SupportSQLiteDatabase database) {
            // Define your migration strategy if needed
            // database.execSQL("ALTER TABLE product ADD COLUMN new_column TEXT");
        }
    };
}

File: app/src/main/java/com/example/shopstock/ProductDB.kt
package com.example.shopstock

import android.content.Context
import androidx.room.Room
import androidx.room.RoomDatabase

abstract class ProductDB : RoomDatabase() {
    abstract fun productDao(): ProductDao
    companion object {
        @Volatile
        private var INSTANCE: ProductDB? = null

        fun getDatabase(context: Context): ProductDB{
            return INSTANCE?: synchronized(this){
                val instance = Room.databaseBuilder(
                    context.applicationContext,
                    ProductDB::class.java,
                    "product_database"
                ).build()
                INSTANCE = instance
                instance
            }
        }
    }
}
